File: testdata/java/overloaded_methods_different_types.java
class Calculator {
    public int add(int a, int b) {
        return a + b;
    }
    
    public double add(double a, double b) {
        return a + b;
    }
    
    public String add(String a, String b) {
        return a + b;
    }
    
    public void test() {
        int x = this.add(1, 2);
        double y = this.add(1.0, 2.0);
        String z = this.add("Hello", "World");
    }
}

File: testdata/go/overloaded_methods_different_types.go
package converted

type calculator struct {
}

func newCalculator() calculator {
	this := calculator{}
	return this
}

func (this *calculator) Add(a int, b int) int {
	return (a + b)
}

func (this *calculator) AddWithFloat64Float64(a float64, b float64) float64 {
	return (a + b)
}

func (this *calculator) AddWithStringString(a string, b string) string {
	return (a + b)
}

func (this *calculator) Test() {
	// FIXME: more than one possible method for add with 2 arguments

	x := this.Add(1, 2)
	// FIXME: more than one possible method for add with 2 arguments

	y := this.Add(1.0, 2.0)
	// FIXME: more than one possible method for add with 2 arguments

	z := this.Add("Hello", "World")
}
